src/psyclone/psyGen.py

```diff
@@ -3908,3 +3913,543 @@ def coloured_text(self):
+            self.process_nodes(parent=array, nodes=subscript_list,
+                               nodes_parent=node.items[1])
+        else:
+            # When just 1 dimension fparser does not have a Subscript_List
+            self.process_nodes(parent=array, nodes=[node.items[1]],
+                               nodes_parent=node)
+
+        return array
+
+    def _number_handler(self, node, parent):
+        '''
+        Transforms an fparser2 NumberBase to the PSyIRe representation.
+
+        :param child: node in fparser2 AST.
+        :type child:  :py:class:`fparser.two.utils.NumberBase`
+        :param parent: Parent node in the PSyclone IR we are constructing.
+        :type parent: :py:class:`psyclone.psyGen.Node`
+
+        :rtype :py:class:`psyclone.psyGen.Literal`
+        '''
+        return Literal(node.items[0], parent=parent)
+
+
+class CodeBlock(Node):
+    '''
+    Node representing some generic Fortran code that PSyclone
+    does not attempt to manipulate. As such it is a leaf in the PSyclone
+    IR and therefore has no children.
+
+    :param statements: list of fparser2 AST nodes representing the Fortran \
+                       code constituting the code block.
+    :type statements: list of :py:class:`fparser.two.utils.Base` or \
+                      :py:class:`fparser.two.utils.BlockBase` objects.
+    :param parent: the parent node of this code block in the PSyIRe.
+    :type parent: :py:class:`psyclone.psyGen.Node`
+    '''
+    def __init__(self, statements, parent=None):
+        super(CodeBlock, self).__init__(parent=parent)
+        # Store a list of the parser objects holding the code associated
+        # with this block. We make a copy of the contents of the list because
+        # the list itself is a temporary product of the process of converting
+        # from the fparser2 AST to the PSyIRe.
+        self._statements = statements[:]
+
+    @property
+    def coloured_text(self):
+        '''
+        Return the name of this node type with control codes for
+        terminal colouring.
+
+        :return: Name of node + control chars for colour.
+        :rtype: str
+        '''
+        return colored("CodeBlock", SCHEDULE_COLOUR_MAP["CodeBlock"])
+
+    def view(self, indent=0):
+        '''
+        Print a representation of this node in the schedule to stdout.
+
+        :param int indent: level to which to indent output.
+        '''
+        print(self.indent(indent) + self.coloured_text + "[" +
+              str(list(map(type, self._statements))) + "]")
+
+    def __str__(self):
+        return "CodeBlock[{0} statements]".format(len(self._statements))
+
+    def gen_code(self):
+        '''
+        Override abstract method from base class.
+
+        :raises InternalError: because it is not relevant to the NEMO API and \
+                               should never be called.
+        '''
+        raise InternalError("CodeBlock.gen_code() should not be called.")
+
+
+class Assignment(Node):
```

Comment:
I ported this from the NemoCodeBlock that had gen_code specialised to produce InternalError instead of the Node gen_code that produces NotImplementedError. I think NonImplementedError is the right one in this context. Therefore I deleted the specialisation and the corresponding test.

Reply:
OK.